src/stirling/source_connectors/socket_tracer/bcc_bpf/protocol_inference.h

```diff
@@ -601,6 +606,32 @@ static __inline enum message_type_t infer_mux_message(const char* buf, size_t co
     return kUnknown;
   }
 
+  if (mux_type == kTdispatch) {
+    // The first context key will start after the header, type, tag, # context fields and first
+    // context key length.  See src/stirling/source_connectors/socket_tracer/protocols/mux/types.h
+    // for a description of the complete frame layout.
+    int first_ctx_pos = 12;
+
+    // While mux is flexible enough to support arbitrary context keys, finagle is
+    // the only implementation of the protocol and Twitter's context fields
+    // com.twitter.finagle.tracing.{TraceContext, ClientIdContext, Deadline} are
+    // the only ones that exist today.
+    if (buf[first_ctx_pos] != 'c' || buf[first_ctx_pos + 1] != 'o' ||
+        buf[first_ctx_pos + 2] != 'm' || buf[first_ctx_pos + 3] != '.' ||
+        buf[first_ctx_pos + 4] != 't' || buf[first_ctx_pos + 5] != 'w' ||
+        buf[first_ctx_pos + 6] != 'i' || buf[first_ctx_pos + 7] != 't' ||
+        buf[first_ctx_pos + 8] != 't' || buf[first_ctx_pos + 9] != 'e' ||
+        buf[first_ctx_pos + 10] != 'r')
+      return kUnknown;
+  }
```

Comment:
This doesn't seem vendor neutral to me?

Reply:
While mux is an open source protocol, it was developed by Twitter and it's only implementation is [finagle](https://github.com/twitter/finagle).

I understand that's not ideal, but the likelihood of mux having an additional set of well known context keys to support is very low.

We must parse [twitter](https://github.com/pixie-io/pixie/blob/2ba99cba01d2d9f4a5f7ffc91e2fafacba62999b/src/stirling/source_connectors/socket_tracer/protocols/mux/parse.cc#L81) specific [context](https://github.com/pixie-io/pixie/blob/2ba99cba01d2d9f4a5f7ffc91e2fafacba62999b/src/stirling/source_connectors/socket_tracer/protocols/mux/parse.cc#L74) keys [in](https://github.com/pixie-io/pixie/blob/2ba99cba01d2d9f4a5f7ffc91e2fafacba62999b/src/stirling/source_connectors/socket_tracer/protocols/mux/parse.cc#L92) order to access certain interesting protocol fields.